### PR TITLE
WT-8474 review clang analyzer warnings

### PR DESCRIPTION
**Pull request by keithbostic**
This branch fixes all but a few of the clang-scan warnings (there's another branch for WT-8466 that moves some files around and fixes two that remain, the ones in `bt_read.c`  and `bt_io.c`).

I did a commit per file, hopefully that makes it more obvious what got changed.

### CODE REVIEW

File: src/evict/evict_page.c [hunk 1]
```diff
@@ -205,7 +205,7 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
     else
         WT_ERR(__evict_page_dirty_update(session, ref, flags));
 
-    if (LF_ISSET(WT_EVICT_CALL_URGENT)) {
+    if (time_start != 0 /* WT_EVICT_CALL_URGENT */) {
```
Comment (luke-pearson) [inline src/evict/evict_page.c:208]: This comment seems out of place, `WT_EVICT_CALL_URGENT` probably left by accident?

File: src/evict/evict_page.c [hunk 2]
```diff
@@ -233,7 +233,7 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
         if (!closing)
             __evict_exclusive_clear(session, ref, previous_state);
 
-        if (LF_ISSET(WT_EVICT_CALL_URGENT)) {
+        if (time_start != 0 /* WT_EVICT_CALL_URGENT */) {
```
Comment (luke-pearson) [inline src/evict/evict_page.c:236]: Same here.

File: src/log/log_slot.c [hunk 3]
```diff
@@ -545,6 +544,8 @@ __wt_log_slot_join(WT_SESSION_IMPL *session, uint64_t mysize, uint32_t flags, WT
         unbuffered = true;
         F_SET(myslot, WT_MYSLOT_UNBUFFERED);
     }
+
+    time_start = 0;
```
Comment (luke-pearson) [inline src/log/log_slot.c:548]: Why move initialization to here?

File: src/evict/evict_page.c (hunk 1 above)
Comment (keithbostic) [reply src/evict/evict_page.c:208]: I did that deliberately --  this is the one change where it wasn't just using `time_start != 0` to turn on timings, but it's using `time_start` as a test for other things, which felt ugly. Thoughts?

I could split stuff apart:```
if (WT_EVICT_CALL_URGENT) {
}
if (time_start) {
}
```
it's not like it's expensive... I'll do that for clarity.

File: src/evict/evict_page.c (hunk 2 above)
Comment (keithbostic) [reply src/evict/evict_page.c:236]: Answered in the previous, related comment.

File: src/log/log_slot.c (hunk 3 above)
Comment (keithbostic) [reply src/log/log_slot.c:548]: It's not needed before then, that is, initialize close to the first use of the variable?

File: src/evict/evict_page.c (hunk 1 above)
Comment (keithbostic) [reply src/evict/evict_page.c:208]: @luke-pearson, I tried this, and it's worse -- I'll just remove the comment as you suggested, it's confusing.

File: src/log/log_slot.c (hunk 3 above)
Comment (luke-pearson) [reply src/log/log_slot.c:548]: I always thought our style was to initialize next to where we declare.

Comment (keithbostic) [reply src/log/log_slot.c:548]: Yes and no? I mean, `for (i = 0;` isn’t initialized next to where we declare? I mean, you’re not wrong — I’ll swap it back.

Comment (keithbostic) [reply src/log/log_slot.c]: Fixed in 5e2e2a6e4.

File: src/evict/evict_page.c (hunk 1 above)
Comment (keithbostic) [reply src/evict/evict_page.c]: Fixed in f395b6935.

File: src/evict/evict_page.c [hunk 4]
```diff
@@ -124,6 +123,7 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
      * Track how long forcible eviction took. Immediately increment the forcible eviction counter,
      * we might do an in-memory split and not an eviction, which skips the other statistics.
      */
+    time_start = 0;
```
Comment (luke-pearson) [inline src/evict/evict_page.c:126]: The same goes for this initialization it feels out of style.

**Review by luke-pearson — APPROVED** [review]
LGTM, with one nitpick comment.

File: src/evict/evict_page.c (hunk 4 above)
Comment (keithbostic) [reply src/evict/evict_page.c:126]: I disagree strongly on this one -- we're swapping out tests of `LF_ISSET(WT_EVICT_CALL_URGENT)` for tests of `time_start`, and I want it very, very clear how that works.

Comment (keithbostic) [reply src/evict/evict_page.c:126]: If you look farther down, there's this:
```
    if (time_start != 0) {
        time_stop = __wt_clock(session);
        if (force_evict_hs)
            WT_STAT_CONN_INCR(session, cache_eviction_force_hs_success);
```
and `time_start` has nothing to do with `force_evict_hs`. This is the one chunk of this change I don't like where we ended up.

Comment (luke-pearson) [reply src/evict/evict_page.c:126]: Happy to leave it as is, I'm not fussed about this. 👍

Comment (luke-pearson) [reply src/evict/evict_page.c:126]: Yeah that `force_evict_hs` definitely feels out of place.

File: cmake/strict/strict_flags_helpers.cmake [hunk 5]
```diff
@@ -169,6 +169,7 @@ function(get_clang_base_flags flags)
         list(APPEND clang_flags "-Wno-unused-command-line-argument")
     endif()
 
+    # FIXME-WT-8052: Figure out whether we want to disable these or change the code.
```
Comment (sueloverso) [inline cmake/strict/strict_flags_helpers.cmake:172]: @etienneptl will a file in this directory be caught by your FIXME scripts?

**Review by sueloverso — APPROVED** [review]
One minor comment about the FIXME comment for Etienne. Otherwise LGTM

File: cmake/strict/strict_flags_helpers.cmake (hunk 5 above)
Comment (etienneptl) [reply cmake/strict/strict_flags_helpers.cmake:172]: Yes, it should! I did run a quick test:
```
> ./s_mentions 
There are comments mentioning wt-8052 in the code, please check if they need to be resolved:
../build_posix/aclocal/strict.m4:158:	# FIXME-WT-8052: Figure out whether we want to disable these or change the code.
../configure:17516:	# FIXME-WT-8052: Figure out whether we want to disable these or change the code.
```

**Pull request merged**